### ISSUE TITLE
Add `--depth=1` param to `git checkout` steps

### ISSUE THREAD

**Issue by srbala** [mgoltzsche/podman-static#23]
Adding `--depth=1` param to `git checkout` steps results in faster checkout and shorter build time. @mgoltzsche if you agree, I can work on a PR.

Current checkout log; `82MB` download
```sh
$ git clone -c 'advice.detachedHead=false' --branch v3.3.1 https://github.com/containers/podman p2 
Cloning into 'pod1'...
remote: Enumerating objects: 129375, done.
remote: Counting objects: 100% (102/102), done.
remote: Compressing objects: 100% (90/90), done.
remote: Total 129375 (delta 42), reused 18 (delta 6), pack-reused 129273
Receiving objects: 100% (129375/129375), 82.13 MiB | 1.50 MiB/s, done.
Resolving deltas: 100% (91638/91638), done.
Updating files: 100% (6192/6192), done.
```
Proposed `--depth=1 ` param checkout log; `11MB` download
```sh
$ git clone -c 'advice.detachedHead=false' --depth=1 --branch v3.3.1 https://github.com/containers/podman p1 Cloning into 'pod2'...
remote: Enumerating objects: 7139, done.
remote: Counting objects: 100% (7139/7139), done.
remote: Compressing objects: 100% (5992/5992), done.
remote: Total 7139 (delta 1035), reused 3617 (delta 630), pack-reused 0
Receiving objects: 100% (7139/7139), 10.80 MiB | 2.05 MiB/s, done.
Resolving deltas: 100% (1035/1035), done.
Updating files: 100% (6192/6192), done.
```

**Comment by mgoltzsche** [comment]
@srbala good point! A PR would be welcome.

**Comment by srbala** [comment]
@mgoltzsche PR ready for review

**Comment by mgoltzsche** [comment]
This was solved with #24.
